src/meta/src/hummock/manager/mod.rs

```diff
@@ -1429,130 +1442,118 @@ impl HummockManager {
+                add_prost_table_stats_map(&mut version_stats.table_stats, &task.table_stats_change);
+                trigger_local_table_stat(
+                    &self.metrics,
+                    &mut versioning.local_metrics,
+                    &version_stats,
+                    &task.table_stats_change,
+                );
             }
+            tasks.push(compact_task);
         }
+        if success_count > 0 {
+            commit_multi_var!(
+                self.env.meta_store(),
+                self.sql_meta_store(),
+                compact_statuses,
+                compact_task_assignment,
+                hummock_version_deltas,
+                version_stats
+            )?;
+            branched_ssts.commit_memory();
+
+            trigger_version_stat(&self.metrics, &current_version);
+            trigger_delta_log_stats(&self.metrics, versioning.hummock_version_deltas.len());
+            self.notify_stats(&versioning.version_stats);
+            versioning.current_version = current_version;
 
-        let task_status = compact_task.task_status();
-        let task_status_label = task_status.as_str_name();
-        let task_type_label = compact_task.task_type().as_str_name();
+            if !deterministic_mode {
+                self.notify_version_deltas(versioning, last_version_id);
+            }
 
-        let label = if is_trivial_reclaim {
-            "trivial-space-reclaim"
-        } else if is_trivial_move {
-            "trivial-move"
+            self.metrics
+                .compact_task_batch_count
+                .with_label_values(&["batch_report_task"])
+                .observe(success_count as f64);
         } else {
+            // The compaction task is cancelled or failed.
+            commit_multi_var!(
+                self.env.meta_store(),
+                self.sql_meta_store(),
+                compact_statuses,
+                compact_task_assignment
+            )?;
+        }
+        let mut success_groups = vec![];
+        for compact_task in tasks {
+            let task_status = compact_task.task_status();
+            let task_status_label = task_status.as_str_name();
+            let task_type_label = compact_task.task_type().as_str_name();
+
             self.compactor_manager
                 .remove_task_heartbeat(compact_task.task_id);
-            "normal"
-        };
-
-        self.metrics
-            .compact_frequency
-            .with_label_values(&[
-                label,
-                &compact_task.compaction_group_id.to_string(),
-                task_type_label,
-                task_status_label,
-            ])
-            .inc();
 
-        tracing::trace!(
-            "Reported compaction task. {}. cost time: {:?}",
-            compact_task_to_string(&compact_task),
-            start_time.elapsed(),
-        );
+            self.metrics
+                .compact_frequency
+                .with_label_values(&[
+                    "normal",
```

Comment:
we can not track the trivial-move task report freq from the metrics "compact_frequency" after this pr ?

Reply:
No, you can check it in L1041